commons/src/main/java/io/hyscale/commons/models/ComponentContext.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.commons.models;
+
+public abstract class ComponentContext {
+
+    private String appName;
+    private String envName;
+    private String serviceName;
+    private String namespace;
```

Comment:
namespace can be in deployment context right ?  It is not used in any of the component contexts.